src/SdfEntityCreator.cc

```diff
@@ -788,8 +790,33 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Visual *_visual)
   // \todo(louise) Populate with default material if undefined
   if (_visual->Material())
   {
+    sdf::Material visualMaterial = *_visual->Material();
+    if (!_visual->Material()->ScriptUri().empty())
+    {
+      std::string scriptUri = visualMaterial.ScriptUri();
+      if (scriptUri != "file://media/materials/scripts/gazebo.material") {
+        gzwarn << "Custom material scripts are not supported."
+          << std::endl;
+      }
+    }
+    if (!_visual->Material()->ScriptName().empty())
+    {
+      std::string scriptName = visualMaterial.ScriptName();
+
+      if((scriptName.find("Gazebo/") != std::string::npos)) {
+        gzwarn << "Using internal gazebo.material to parse " << scriptName
+          << std::endl;
+        MaterialParser* materialParser = new MaterialParser();
```

Comment:
This can be a member of `SdfEntityCreatorPrivate` so we don't initialize it for each visual. It also doesn't need to be a pointer.

Reply:
The `MaterialParser.hh` isn't to be installed, so can't directly include, how can I access the header?

Reply:
That shouldn't be a problem. It's already included in line 87 and working fine. And you won't need to change that since `SdfEntityCreatorPrivate` is defined in this file.